skimage/feature/texture.py

```diff
@@ -97,11 +97,31 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
+
+    #if image.dtype in [np.float, np.float16, np.float32, np.float]
+    if image.dtype == np.float:
+        raise ValueError("Float images are not supported by greycomatrix."
+                         "The image needs to be casted to an unsigned integer type.")
+    
+    # for 16 bit images, levels must be set.
+    if image.dtype != np.uint8:
+        if levels is None:
+            raise ValueError("For images other than 8 bit images, argument levels must be set to a user defined value."
+                             "A large value will produce a large output matrix, potentially causing memory problems."
+                             "In this case, one might consider binning the image prior to calling greycomatrix.")
```

Comment:
All code must fit in 80 characters. This message probably needs to be made a bit more concise, e.g.:

"The `levels` argument is required for data types other than uint8. "
"The resulting matrix will be at least `levels ** 2` in size."

If you want to prescribe the solution to users, I would suggest doing so in a note in the docstring, rather than in the exception.